### PR TITLE
Add C implementation for `@stdlib/math/base/special/acoth`

### LINKED ISSUE #756: [RFC]: Add C implementation for `@stdlib/math/base/special/acoth`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/acoth`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acoth).

```c
double stdlib_base_acoth( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #756.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/native/benchmark.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+double benchmark() {
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 100.0*rand_double() ) + 1.1;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/native/benchmark.c:104]: I copied over this file from `c/benchmark.c` is this assignment correct?

Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/native/benchmark.c:104]: I copied over this file from `c/benchmark.c`, not sure if this assignment is correct.

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/native/benchmark.c (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/acoth/benchmark/c/native/benchmark.c:104]: It is correct, as `acoth` is not defined on the open interval `(-1, 1)`, so the `1.1` is a small offset to ensure a valid lower bound.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**